core/templates/pages/exploration-player-page/services/prediction-algorithm-registry.service.ts

```diff
@@ -16,46 +16,61 @@
  * @fileoverview Service for mapping algorithmId to PredictionAlgorithmService.
  */
 
-angular.module('oppia').factory('PredictionAlgorithmRegistryService', [
-  '$injector', function($injector) {
-    /**
-     * This mapping needs to be updated whenever a new prediction service needs
-     * to be added for classification. The mapping is from algorithmId to a
-     * list of objects. The mapping should be of the type:
-     * {
-     *   algorithmId: {
-     *     dataSchemaVersion: predictionService
-     *   }
-     * }
-     */
-    var algorithmIdPredictionServiceMapping = {
-      CodeClassifier: {
-        v1: 'CodeReplPredictionService'
-      },
-      TextClassifier: {
-        v1: 'TextInputPredictionService'
-      }
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { CodeReplPredictionService } from
+  'interactions/CodeRepl/code-repl-prediction.service';
+import { TextInputPredictionService } from
+  'interactions/TextInput/text-input-prediction.service';
+
+interface IPredictionService {
+  predict(classifierData, answer): number;
+}
+
+type AlgorithmIdPredictionServiceMap =
```

Comment:
The preferred pattern is to wrap statements in `(...)` when it spans multiple lines and have it break after `(` (see coding style [here](https://github.com/oppia/oppia/wiki/Coding-style-guide#javascript)).

Reply:
Done.